java/adapter/avro/src/main/java/org/apache/arrow/AvroToArrowUtils.java

```diff
@@ -113,7 +113,13 @@ private static ArrowType getArrowType(Type type) {
     schema.getObjectProps().forEach((k,v) -> metadata.put(k, v.toString()));
 
     if (type == Type.RECORD) {
-      throw new UnsupportedOperationException();
+      for (Schema.Field field : schema.getFields()) {
+        final ArrowType arrowType = getArrowType(field.schema().getType());
+        final FieldType fieldType = new FieldType(/*nullable=*/true, arrowType, /*dictionary=*/null, /*metadata=*/null);
```

Comment:
it can be a follow-up PR but each field appears to have getObjectProps as well.

Reply:
Thanks, I fixed now.